src/ansys/fluent/core/workflow.py

```diff
@@ -1183,18 +1189,20 @@ def add_child_and_update(self, state=None, defer_update=None):
             )
         self._command_source._compound_child = True
         self._command_source._parent_of_compound_child = py_name
-        if self._fluent_version >= FluentVersion.v241:
-            if defer_update is None:
-                defer_update = False
-            self._task.AddChildAndUpdate(DeferUpdate=defer_update)
-        else:
-            if defer_update is not None:
-                warnings.warn(
-                    " The 'defer_update()' method is supported in Fluent 2024 R1 and later.",
-                    PyFluentUserWarning,
-                )
-            self._task.AddChildAndUpdate()
-        self._command_source._compound_child = False
+        try:
+            if self._fluent_version >= FluentVersion.v241:
+                if defer_update is None:
+                    defer_update = False
+                self._task.AddChildAndUpdate(DeferUpdate=defer_update)
+            else:
+                if defer_update is not None:
+                    warnings.warn(
+                        " The 'defer_update()' method is supported in Fluent 2024 R1 and later.",
```

Comment:
It's an argument, rather than a method, and it's supported _with_ that version of Fluent rather than in it. I also think it would be useful if the string conversion of the given `FluentVersion` had the required format to be used everywhere we want to output it as a string.

Reply:
Is the leading space required?

Reply:
> Is the leading space in the warning message required?

No, will remove it.

Reply:
Is this fine:
"The 'defer_update' argument is supported with 'FluentVersion.v242' and later."

or

"The 'defer_update' argument is supported with Fluent version '24.2.0' and later."

Reply:
The way you wrote the Fluent version was correct in the first place, so don't change that. I was only suggesting that we could in general be automating that based on the enum so that we would always give the output in a consistent format. Otherwise this change is fine.